Add inductor_fx_graph_cache stats to dynamo_utils

ppanchalia opened the pull request:
Summary:
Add the following inductor fx graph cache stats to dynamo compile

- inductor_fx_cache_hit_count
- inductor_fx_cache_miss_count
- inductor_fx_cache_backend_type
- inductor_fx_cache_hit_keys
- inductor_fx_cache_miss_keys
- remote_cache_version

Test Plan: Run local tests and staging logger: P1683061460

Differential Revision: D66232206




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

ppanchalia commented:
@pytorchbot label "topic: not user facing"

masnesral commented on torch/_dynamo/utils.py:
```diff
@@ -961,8 +967,33 @@ def us_to_ms(field):
+        metric = metrics.get(field, None)
+        if metric is None:
+            return None
+        del metrics[field]
```
Is this needed? I can't think of a reason why.

ppanchalia replied:
Since we are passing this onto CompilationMetrics, I am just scared but the clashes that `field` can have on the value (one set the other str).

masnesral replied:
Ok, here's my vote for the "clean" solution I think we should do this in the long run:
* The fields in CompilationMetrics should be declared as set type (or list or dict if we have future fields of those types).
* We should not do conversion to comma-separated strings in this file. The fields passed in the metrics param should just flow unmodified into the CompilationMetrics object we create here.
* `log_compilation_event()` is the function that knows how to log a CompilationMetrics object to scuba. So it should do any reformatting of CompilationMetrics fields before logging, including turning any set fields into comma-separated strings. But doing that change is kind of a PITA because we need to also modify an fbcode-only file. So this seems fine for now; let's not block this PR. Plus I don't know if you'd even agree with me, lol

ppanchalia replied:
@masnesral  -- I agree with you, we can decouple the small-ish refactor to a different PR/diff.

masnesral commented on torch/_dynamo/utils.py:
```diff
@@ -961,8 +967,33 @@ def us_to_ms(field):
+        return ",".join(safe_str(item) for item in metric)
+
+    # Convert sets to strings
+    inductor_fx_remote_cache_hit_keys_str = _convert_collection_to_str(
```
nit: Completely stylistic, so hardly worth commenting, but why not just put the call to your helper directly below?

```
"inductor_fx_remote_cache_hit_keys": _convert_collection_to_str(
```

ppanchalia committed:
Add inductor_fx_graph_cache stats to dynamo_utils (#141190)

Summary:

Add the following inductor fx graph cache stats to dynamo compile

- inductor_fx_cache_hit_count
- inductor_fx_cache_miss_count
- inductor_fx_cache_backend_type
- inductor_fx_cache_hit_keys
- inductor_fx_cache_miss_keys
- remote_cache_version

Test Plan: Run local tests and staging logger: P1683061460

Differential Revision: D66232206

ppanchalia commented:
@masnesral -- I agree with you, we can decouple the small-ish refactor to a different PR/diff.

ppanchalia commented:
@pytorchbot merge

izaitsevfb commented:
@pytorchbot merge
